src/plugins/Library/index/TermEntry.java

```diff
@@ -37,7 +37,8 @@
 
 	public TermEntry(String s, float r) {
 		if (s == null) {
-			throw new IllegalArgumentException("can't have a null subject!");
+//			throw new IllegalArgumentException("can't have a null subject!");
+			s = "null"; // FIXME
```

Comment:
Probably related to https://github.com/freenet/plugin-Library/pull/16/files#diff-aa254c0343e62a1fcd16f7077a106f09R302. Not sure if that gives away a bit more of context.

Reply:
As far as I can remember this situation, this is about the fact that the subject can contain the human word "null", but then it turns into `null` (smart library mode)

Reply:
That sounds like something that should be fixed in a different place… but as with the rest of this PR I’m totally not qualified to talk about any of it because I have no clue of all the contexts in question. :)